test/extended/etcd/vertical_scaling.go

```diff
@@ -112,4 +112,78 @@ var _ = g.Describe("[sig-etcd][Serial] etcd [apigroup:config.openshift.io]", fun
 		err = errors.Wrap(err, "scale-down: timed out waiting for only 3 Running master machines")
 		o.Expect(err).ToNot(o.HaveOccurred())
 	})
+
+	// The following test covers replacing a failed master machine from 3 master nodes cluster.
+	// It starts by setting one of the master machine to be failed/unhealthy.
+	// The failed machine is expected to be replaced automatically.
+	// The test ends validating the size of the cluster, and asserting the member was removed from the etcd cluster by contacting MemberList API.
+	g.It("is able to replace failed master machine from 3 master nodes cluster [Timeout:60m][apigroup:machine.openshift.io]", func() {
+		// set up
+		// TODO: shall we refactor this into `BeforeEach()`
+		ctx := context.TODO()
+		etcdClientFactory := scalingtestinglibrary.NewEtcdClientFactory(oc.KubeClient())
+		machineClientSet, err := machineclient.NewForConfig(oc.KubeFramework().ClientConfig())
+		o.Expect(err).ToNot(o.HaveOccurred())
+		machineClient := machineClientSet.MachineV1beta1().Machines("openshift-machine-api")
+		kubeClient := oc.KubeClient()
+
+		// make sure it can be run on the current platform
+		// TODO: shall we refactor this into `JustBeforeEach()`
+		scalingtestinglibrary.SkipIfUnsupportedPlatform(ctx, oc)
```

Comment:
This will only work today on AWS, 4.12 onwards, is this accounting for that?

In later releases we may add additional automated set up, eg for Azure and GCP

Reply:
I think so, isn't it @hasbro17

Reply:
I added `OpenStack` and `GCP` to the skipped jobs @JoelSpeed

Reply:
No it won't. You need to detect whether the platform supports the `cluster` CPMS and then skip. Responded in https://github.com/openshift/origin/pull/27496#discussion_r1009884240